providers/implementations/kdfs/hkdf.c

```diff
@@ -426,6 +447,14 @@ static int HKDF_Extract(OSSL_LIB_CTX *libctx, const EVP_MD *evp_md,
                         unsigned char *prk, size_t prk_len)
 {
     int sz = EVP_MD_get_size(evp_md);
+    /*
+     * Set pedantic to zero to allow a salt with arbitrary length.
+     */
+    int pedantic = 0;
+    OSSL_PARAM mac_params[] = {
```

Comment:
Unfortunately, this won't build for some platforms which consider &pedantic to not be constant.

Does this impact FIPS compliance?

Reply:
Fixed.

Our lab think it only need to ensure that HMAC function and key are strong enough. Therefore, I think the length of salt should not impact the compliance.

Reply:
You misunderstood.  Taking the address of `pedantic` is not considered a constant initialiser on some platforms.  I.e. this code will not compile.

This is why we use other methods to construct OSSL_PARAM arrays at runtime.

Reply:
The method to construct OSSL_PARAM arrays changed.